src/fasta/fasta.cpp

```diff
@@ -88,6 +102,7 @@ class ebi::vcf::fasta::ContigFromENA {
   std::string read(const size_t pos, const size_t length)
   {
       if (!fasta_input.get()) {
+          fasta_output->flush();
```

Comment:
I'm wondering if all this details could be avoided if we simplified the design of these classes. Is there any reason not to make `fasta_input` and `fasta_output` regular fstreams instead of pointers? that way they would be initialized in the constructor and destroyed automatically in the destructor.

Reply:
This one was not because of using pointer for `fasta_output`. It was caused by separating output and input streams for the same underlying temporary file. I wanted to use a same `fstream` object for writting the downloaded fasta to and reading the sequence from. But at that point, I could not get it working.
I have revisited this issue and now found a proper way to open a `fstream` for both input and output, as changed in d599e3b. Also refactor other places that use smart pointer unnecessarily.
Thank you very much for pointing this out. Please let me know anything further I can do.